apps/epic-web/src/templates/embed-template.tsx

```diff
@@ -77,7 +81,11 @@ const EmbedTemplate: React.FC<VideoEmbedPageProps> = ({
         }}
       />
       <ModuleProgressProvider moduleSlug={module.slug.current}>
-        <LessonProvider lesson={lesson} module={module} section={section}>
+        <LessonProvider
+          lesson={isSolution ? {...solution, slug: lesson.slug} : lesson}
```

Comment:
this is what we do on the non-embed page

the problem is a solution isn't a "real" lesson so the query and zod schema fails which makes the CASL abilities lookup fail